python/lsst/ts/ess/config_schema.py

```diff
@@ -25,57 +25,59 @@
 
 CONFIG_SCHEMA = yaml.safe_load(
     """
-    $schema: http://json-schema.org/draft-07/schema#
-    $id: https://github.com/lsst-ts/ts_ess/blob/master/python/lsst/ts/ess/config_schema.py
-    # title must end with one or more spaces followed by the schema version, which must begin with "v"
-    title: ESS v1
-    description: Schema for ESS configuration files
-    type: object
-    properties:
-      host:
-        description: IP address of the TCP/IP interface
-        type: string
-        format: hostname
-        default: "127.0.0.1"
-      port:
-        description: Port number of the TCP/IP interface
-        type: integer
-        default: 5000
-      name:
-        description: Name of the sensor
-        type: string
-        default: EssTemperature4Ch
-      channels:
-        description: Number of channels
-        type: integer
-        default: 4
-      type:
-        description: Type of the device
-        type: string
-        enum:
-        - FTDI
-        - Serial
-        default: FTDI
-    anyOf:
-    - if:
-        properties:
-          type:
-            const: FTDI
-      then:
-        properties:
-          ftdi_id:
-            description: FTDI Serial ID to connect to.
-            type: string
-            default: "AL05OBVR"
-    - if:
-        properties:
-          type:
-            const: Serial
-      then:
-        properties:
-          port:
-            description: Serial port to connect to.
-            type: string
-            default: "serial_ch_1"
-    """
+$schema: http://json-schema.org/draft-07/schema#
+$id: https://github.com/lsst-ts/ts_ess/blob/master/python/lsst/ts/ess/config_schema.py
+# title must end with one or more spaces followed by the schema version, which must begin with "v"
+title: ESS v1
+description: Schema for ESS configuration files
+type: object
+properties:
+  host:
+    description: IP address of the TCP/IP interface
+    type: string
+    format: hostname
+    default: "127.0.0.1"
+  port:
+    description: Port number of the TCP/IP interface
+    type: integer
+    default: 5000
+  devices:
+    type: array
+    default: [{"name": "EssTemperature4Ch", "channels": 4, "type": "FTDI", "ftdi_id": "AL05OBVR"}]
+    items:
+      type: object
+      properties:
+        name:
+          description: Name of the sensor
+          type: string
+        channels:
+          description: Number of channels
+          type: integer
+          default: 4
```

Comment:
I gently suggest removing this default, because salobj.DefaultingValidator can't use defaults deeper than 2 levels.

If you want to suggest 4 as a value to users you could add that information to the `description`, possibly with an explanation.

Reply:
Thanks. This one slipped past my attention when I was removing the default values. Done.